packages/waku/src/router/create-pages.ts

```diff
@@ -180,11 +178,11 @@ export type CreateLayout = <T extends string>(layout: {
   >;
 }) => void;
 
-export function createPages(
+export function createPages<AllPages extends AnyPage[]>(
   fn: (fns: {
     createPage: CreatePage;
     createLayout: CreateLayout;
-  }) => Promise<void>,
+  }) => Promise<AllPages | void>,
```

Comment:
Is `void` for backward compatibility? I wonder if it breaks so much.

Reply:
Yes it is explicitly for backwards compatibility. 

If we made a breaking change for this that would simplify our code a bit and encourage the move to type-safe paths. The breaking change is also a very easy fix for users.

I'm happy with whichever you'd prefer though.

Reply:
I wouldn't expect many people use `createPages` even for experiments.
This is type only breaking change, so it's reasonable. Let's simplify the code.

Reply:
Removed this ✅